cli/cmd/report_rules.go

```diff
@@ -0,0 +1,409 @@
+		},
+		{
+			Name: "channels",
+			Prompt: &survey.MultiSelect{
+				Message: "Select email alert channels:",
+				Options: channelList,
+			},
+			Validate: survey.Required,
+		},
+		{
+			Name: "severities",
+			Prompt: &survey.MultiSelect{
+				Message: "Select severities:",
+				Options: []string{"Critical", "High", "Medium", "Low", "Info"},
+			},
+		},
+		{
+			Name: "notifications",
+			Prompt: &survey.MultiSelect{
+				Message: "Select report notification types:",
+				Options: notificationFields,
+			},
+		},
+	}
+
+	answers := struct {
+		Name           string
+		Description    string   `survey:"description"`
+		Channels       []string `survey:"channels"`
+		Severities     []string `survey:"severities"`
+		ResourceGroups []string `survey:"resourceGroups"`
+		Notifications  []string `survey:"notifications"`
+	}{}
+
+	err := survey.Ask(questions, &answers,
+		survey.WithIcons(promptIconsFunc),
+	)
+	if err != nil {
+		return api.ReportRuleResponse{}, err
+	}
+
+	var channels []string
+	for _, channel := range answers.Channels {
+		channels = append(channels, channelMap[channel])
+	}
+
+	resourceGroups, resourceGroupMap := promptAddResourceGroupsToReportRule()
+	var groups []string
+	for _, group := range resourceGroups {
+		groups = append(groups, resourceGroupMap[group])
+	}
+
+	for _, n := range answers.Notifications {
+		notificationsMap[n] = true
+	}
+
+	notifications := api.ReportRuleNotificationTypes{}
+	err = api.TransformReportRuleNotification(notificationsMap, &notifications)
+	if err != nil {
+		return api.ReportRuleResponse{}, err
+	}
+
+	reportRule, err := api.NewReportRule(
+		answers.Name,
+		api.ReportRuleConfig{
+			Description:        answers.Description,
+			Severities:         api.NewReportRuleSeverities(answers.Severities),
+			ResourceGroups:     groups,
+			EmailAlertChannels: channels,
+			NotificationTypes:  api.ReportRuleNotifications{notifications},
+		})
+
+	if err != nil {
+		return api.ReportRuleResponse{}, err
+	}
+
+	cli.StartProgress(" Creating report rule...")
+	defer cli.StopProgress()
+
+	return cli.LwApi.V2.ReportRules.Create(reportRule)
+}
+
+func getEmailAlertChannels() ([]string, map[string]string) {
+	response, err := cli.LwApi.V2.AlertChannels.List()
+
+	if err != nil {
+		return nil, nil
+	}
+	var items = make(map[string]string)
+	var channels = make([]string, 0)
+	for _, i := range response.Data {
+		if i.AlertChannelType() == api.EmailUserAlertChannelType {
+			displayName := fmt.Sprintf("%s - %s", i.ID(), i.Name)
+			channels = append(channels, displayName)
+			items[displayName] = i.ID()
+		}
+	}
+
+	return channels, items
+}
+
+func promptAddResourceGroupsToReportRule() ([]string, map[string]string) {
+	addResourceGroups := false
+	err := survey.AskOne(&survey.Confirm{
+		Message: "Add Resource Groups to Report Rule?",
+	}, &addResourceGroups)
+
+	if err != nil {
+		return nil, nil
+	}
+
+	if addResourceGroups {
+		var groups []string
+		groupList, groupMap := getResourceGroups()
```

Comment:
would be nice to also add a progress inside this function, it doesn't have it

Reply:
👍